internal/provider/data_source_hvn_route.go

```diff
@@ -0,0 +1,98 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+func dataSourceHVNRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN Route data source provides information about an existing HVN route.",
+		ReadContext: dataSourceHVNRouteRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"destination_cidr": {
+				Description: "The destination CIDR of the HVN route",
+				Type:        schema.TypeString,
```

Comment:
Thinking of introducing CIDR validation here, but wanted to see what the team thinks before adding it.

Reply:
I'm not sure it's necessary, backend already does the validation

Reply:
Generally, the provider should duplicate any straightforward input validation (i.e. doesn't require an API call) so requests fail early. Just remembered terraform provider SDK provides a [CIDR helper](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/validation/network.go#L84) 🙌